pkg/bootstrap/bootstrap_provider.go

```diff
@@ -554,6 +612,28 @@ func deployKeyName(namespace, secretName, branch, path string) string {
 	return name
 }
 
+// newDeployTokenInfo constructs a gitprovider.DeployTokenInfo with the
+// given values and returns the result.
+func newDeployTokenInfo(name string) gitprovider.DeployTokenInfo {
+	tokenInfo := gitprovider.DeployTokenInfo{Name: name}
+	return tokenInfo
+}
+
+func deployTokenName(namespace, secretName, branch, path string) string {
+	var name string
+	for _, v := range []string{namespace, secretName, branch, path} {
+		if v == "" {
+			continue
+		}
+		if name == "" {
+			name = v
+		} else {
+			name = name + "-" + v
+		}
+	}
+	return name
```

Comment:
How about this instead which is easier to read IMHO:

```suggestion
	var elems []string
	for _, v := range []string{namespace, secretName, branch, path} {
		if v == "" {
			continue
		}
		elems = append(elems, v)
	}
	return strings.Join(elems, "-")
```